src/protocol/main.mo

```diff
@@ -95,6 +95,36 @@ actor Minter {
     }
   };
 
+  public query func getPositions(limit: Nat, offset: Nat): async [P.SharedPosition] {
+    if(offset > lastPositionId or offset < 0) {
+        throw Error.reject("Wrong offset");
+    };
+    if(limit < 0) {
+        throw Error.reject("Wrong limit");
+    };
+    var positions:[P.SharedPosition]=[];
+    var start: Nat = offset;
+    var end: Nat = offset+limit;
+    if(end > lastPositionId) {
+       end := lastPositionId;
+    };
+    if(end > 200) {
+       end := offset + 200;
+    };
+    while(start <= end) {
+        switch(positionMap.get(start)) {
+          case null {
+            throw Error.reject("No position found.");
+          };
+          case (?position) {
+            positions:=Array.append<P.SharedPosition>([P.SharedPosition(position)],  positions);
```

Comment:
Looks like it is better to use Buffer instead Array
https://internetcomputer.org/docs/current/developer-docs/build/languages/motoko/control-flow#using-range-with-a-for-loop

Also you appending in reverse order here